### PR TITLE
Refactor/retry restart exporter

### PR DESCRIPTION
**Pull request by jneo8**

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -125,29 +125,35 @@ def _on_update_status(self, _: EventBase) -> None:  # noqa: C901
...
-                    "Exporter crashed unexpectedly, please refer to systemd logs..."
-                )
+            restart_ok, restart_status = self.restart_exporter()
+            if not restart_ok and restart_status is not None:
+                self.model.unit.status = restart_status
                 return
 
         self.model.unit.status = ActiveStatus("Unit is ready")
 
+    def restart_exporter(self) -> Tuple[bool, Optional[StatusBase]]:
```
Comment (chanchiwai-ray) [inline src/charm.py:135]: It's probably enough to return `Optional[StatusBase]`.

File: src/charm.py [hunk 2]
```diff
@@ -125,29 +125,35 @@ def _on_update_status(self, _: EventBase) -> None:  # noqa: C901
...
-                        "Exporter crashed unexpectedly, please refer to systemd logs..."
-                    )
-                    return
-            except Exception as err:  # pylint: disable=W0718
-                logger.error("Exporter crashed unexpectedly: %s", err)
-                self.model.unit.status = ErrorStatus(
-                    "Exporter crashed unexpectedly, please refer to systemd logs..."
-                )
+            restart_ok, restart_status = self.restart_exporter()
+            if not restart_ok and restart_status is not None:
```
Comment (chanchiwai-ray) [inline src/charm.py:129]: Following from the other comment: `if restart_status is not None:` should be enough

**Review by chanchiwai-ray — CHANGES_REQUESTED** [review]
Small suggestion

**Review by Pjack — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (jneo8) [reply src/charm.py:135]: I choose not to use None to represent its success.

Comment (chanchiwai-ray) [reply src/charm.py:135]: 🆗

**Review by chanchiwai-ray — APPROVED** [review]

**Pull request merged**